rotkehlchen/tests/unit/decoders/test_curve_lend.py

```diff
@@ -0,0 +1,694 @@
+from typing import TYPE_CHECKING
+
+import pytest
+
+from rotkehlchen.accounting.structures.balance import Balance
+from rotkehlchen.assets.asset import EvmToken, UnderlyingToken
+from rotkehlchen.assets.utils import get_or_create_evm_token
+from rotkehlchen.chain.arbitrum_one.modules.curve_lend.decoder import (
+    LLAMALEND_LEVERAGE_ZAP as ARB_LLAMALEND_LEVERAGE_ZAP,
+)
+from rotkehlchen.chain.evm.constants import ZERO_ADDRESS
+from rotkehlchen.chain.evm.decoding.constants import CPT_GAS
+from rotkehlchen.chain.evm.decoding.curve.constants import CPT_CURVE
+from rotkehlchen.chain.evm.types import string_to_evm_address
+from rotkehlchen.constants import ONE
+from rotkehlchen.constants.assets import A_ETH, A_WBTC, A_WETH_ARB
+from rotkehlchen.fval import FVal
+from rotkehlchen.history.events.structures.evm_event import EvmEvent
+from rotkehlchen.history.events.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.tests.utils.ethereum import get_decoded_events_of_transaction
+from rotkehlchen.types import (
+    CURVE_VAULTS_PROTOCOL,
+    ChainID,
+    EvmTokenKind,
+    Location,
+    TimestampMS,
+    deserialize_evm_tx_hash,
+)
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.arbitrum_one.node_inquirer import ArbitrumOneInquirer
+    from rotkehlchen.chain.ethereum.node_inquirer import EthereumInquirer
+    from rotkehlchen.chain.optimism.node_inquirer import OptimismInquirer
+    from rotkehlchen.db.dbhandler import DBHandler
+    from rotkehlchen.types import ChecksumEvmAddress
+
+
+def _create_arbitrum_vault_tokens(database: 'DBHandler') -> tuple['EvmToken', 'EvmToken']:
```

Comment:
For those _create_xxx_vault_tokens() can't you turn them into fixtures and use them in the tests you need them in?

Reply:
Converted them to fixtures, but note that the vault token always needs to be present even if its not referenced in the test, since that is what is used to trigger decoding. So there are some tests where both the vault token and underlying token fixture are used, but only the underlying token is referenced in the test.